### PR TITLE
Warn users about collab editing

### PR DESCRIPTION
**Pull request by Raemon**
* Adds a warning to the sharing dialog that Collaborative Editing is in beta, and that people should message us on intercom if they experience issues
* Removes the "share with users" UI, which was kinda duplicating the "share" button but in a way that didn't make it clear that users were opting into a beta feature.
* Makes the "Share" button less obtrusive, since it's not currently polished enough to highlight as strongly.

(So, now all users, regardless of 'opt into beta' checkbox, experience the same button)

image

image

### CODE REVIEW

File: packages/lesswrong/components/editor/PostSharingSettings.tsx [hunk 1]
```diff
@@ -209,6 +213,8 @@ const PostSharingSettingsDialog = ({postId, linkSharingKey, initialSharingSettin
         </Select>
       </div>
       
+      <p className={classes.warning}>Collaborative editing is in beta. Message us on intercom if you experience issues.</p>
```
Comment (darkruby501) [inline packages/lesswrong/components/editor/PostSharingSettings.tsx:216]:
```suggestion
      <p className={classes.warning}>Collaborative Editing features are in beta. User beware! Message us on Intercom if you experience issues.</p>
```

File: packages/lesswrong/lib/betas.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export const userHasBoldPostItems = disabled
 export const userHasEAHomeHandbook = adminOnly
 export const userCanCreateCommitMessages = moderatorOnly;
 export const userHasRedesignedSettingsPage = disabled;
-export const userCanUseSharing = (user: UsersCurrent|DbUser|null): boolean => (moderatorOnly(user) || (optInOnly(user) && karmaGated(20)(user)))
+export const userCanUseSharing = (user: UsersCurrent|DbUser|null): boolean => (moderatorOnly(user) || karmaGated(20)(user))
```
Comment (darkruby501) [inline packages/lesswrong/lib/betas.ts:31]:
```suggestion
export const userCanUseSharing = (user: UsersCurrent|DbUser|null): boolean => (moderatorOnly(user) || karmaGated(1)(user))
```

**Comment by darkruby501** [comment]
Test Cases:
- New User Form; can click button and add users.


- new post form
- existing post form

- before/after save

* Add new users to doc
* Edit users on doc

**Review by jimrandomh — APPROVED** [review]

File: packages/lesswrong/components/editor/PostSharingSettings.tsx [hunk 3]
```diff
@@ -225,6 +229,8 @@ const PostSharingSettingsDialog = ({postId, linkSharingKey, initialSharingSettin
         </Select>
       </div>
       
+      <p className={classes.warning}>Collaborative Editing features are in beta. Message us on Intercom or email team@lesswrong.com if you experience issues.</p>
```
Comment (darkruby501) [inline packages/lesswrong/components/editor/PostSharingSettings.tsx:232]:
```suggestion
      <p className={classes.warning}>Collaborative Editing features are in beta. Message us on Intercom or email us at team@lesswrong.com if you experience issues</p>
```

**Pull request merged**